algorithms/src/snark/varuna/ahp/indexer/circuit.rs

```diff
@@ -139,12 +139,11 @@ impl<F: PrimeField, SM: SNARKMode> Circuit<F, SM> {
     }
 
     pub fn interpolate_matrix_evals(&self) -> Result<impl Iterator<Item = LabeledPolynomial<F>>> {
-        Ok([("a", &self.a_arith), ("b", &self.b_arith), ("c", &self.c_arith)]
-            .into_iter()
-            .map(|(label, evals)| MatrixArithmetization::new(&self.id, label, evals))
-            .collect::<Result<Vec<_>, _>>()?
-            .into_iter()
-            .flat_map(|arith| arith.into_iter()))
+        let mut iters = Vec::with_capacity(3);
+        for (label, evals) in [("a", &self.a_arith), ("b", &self.b_arith), ("c", &self.c_arith)] {
+            iters.push(MatrixArithmetization::new(&self.id, label, evals)?.into_iter());
+        }
+        Ok(iters.into_iter().flatten())
```

Comment:
@ljedrz please review for performance hit. Call to allocated vector followed by constructing a flattened iterator needs performance review.

Reply:
this one is most likely fine, as we've used `collect` before; I'll run some benches to double-check